### PR TITLE
Adds an action button for succumbing, broadcasts last words

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Adds an action button to succumb when you hit critical health. Inspired by /tg/.
Before you ask, yes. If you click the button by mistake, you can just cancel the last words dialog. If you use the verb, though, you're in it for the long haul.
Also, if a user had some last words, this should share them in chat to all the ghosts listening.

## Why It's Good For The Game
Reminds people who might not have otherwise known that succumbing is an option, and makes it easier to access for everyone else. Also, this makes last words (which might otherwise be out of earshot) a bit more impactful.

## Images of changes
image

## Testing
Beat myself to a pulp, shuffled off this mortal coil, did it again, clicked the button.

## Changelog
:cl:
add: A new action button to succumb will appear when you enter critical health.
add: Last words will now be sent out to dchat aas well.
/:cl:

### CODE REVIEW

File: code/_onclick/hud/alert.dm [hunk 1]
```diff
@@ -817,3 +817,18 @@ so as to remain in compliance with the most up-to-date laws."
 	master = null
 	screen_loc = ""
 	return ..()
+
```
Comment (DGamerL) [inline code/_onclick/hud/alert.dm:820]:
```suggestion
```

File: code/modules/mob/living/carbon/human/human_life.dm [hunk 2]
```diff
@@ -662,6 +662,12 @@
 				healthdoll.overlays -= (cached_overlays - new_overlays)
 				healthdoll.cached_healthdoll_overlays = new_overlays
 
+		if(health <= HEALTH_THRESHOLD_CRIT)
+			throw_alert("succumb", /atom/movable/screen/alert/succumb)
+		else
+			clear_alert("succumb")
+
```
Comment (DGamerL) [inline code/modules/mob/living/carbon/human/human_life.dm:669]:
```suggestion
```

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -322,6 +330,7 @@
 	to_chat(src, "<span class='notice'>You have given up life and succumbed to death.</span>")
 	apply_status_effect(STATUS_EFFECT_RECENTLY_SUCCUMBED)
 
+
```
Comment (DGamerL) [inline code/modules/mob/living/living.dm:333]:
```suggestion
```

**Review by DGamerL — APPROVED** [review]

File: code/_onclick/hud/alert.dm [hunk 4]
```diff
@@ -817,3 +817,17 @@ so as to remain in compliance with the most up-to-date laws."
...
+	desc = "Shuffle off this mortal coil."
+	icon_state = "succumb"
+
+/atom/movable/screen/alert/succumb/Click()
+	if(!usr || !usr.client)
+		return
+	var/mob/living/living_owner = usr
+	if(!istype(usr))
+		return
+	living_owner.do_succumb(TRUE)
```
Comment (SteelSlayer) [inline code/_onclick/hud/alert.dm:833]: Does this alert automatically clear after they die? If not, it could be stuck on their hud while dead/a ghost.

Comment (lewcc) [reply code/_onclick/hud/alert.dm:833]: It seemed to, from my testing. I can double check.

Comment (lewcc) [reply code/_onclick/hud/alert.dm:833]: Fixed

File: code/modules/mob/living/death.dm [hunk 5]
```diff
@@ -87,10 +87,14 @@
 				var/mob/M = P
 				if((M.client?.prefs.toggles2 & PREFTOGGLE_2_DEATHMESSAGE) && (isobserver(M) || M.stat == DEAD))
 					to_chat(M, "<span class='deadsay'><b>[mind.name]</b> has died at <b>[area_name]</b>. (<a href='?src=[M.UID()];jump=\ref[T]'>JMP</a>)</span>")
+					if(last_words)
+						to_chat(M, "<span class='deadsay'><b>[p_their(TRUE)] last words were:<b> \"[last_words]\"</span>")
```
Comment (lewcc) [inline code/modules/mob/living/death.dm:91]:
```suggestion
						to_chat(M, "<span class='deadsay'><b>[p_their(TRUE)] last words were:</b> \"[last_words]\"</span>")
```

**Review by S34NW — APPROVED** [review]

**Pull request merged**